### PR TITLE
add blockNonce cache to TransactionNonceCheck in case of the overhead caused by obtaining old nonces from db

### PR DESCRIPTION
**Pull request by cyjseagull**

### CODE REVIEW

File: libdevcore/BasicLevelDB.cpp [hunk 1]
```diff
@@ -67,14 +67,14 @@ BasicLevelDB::BasicLevelDB(const leveldb::Options& _options, const std::string&
     {
         std::stringstream exitInfo;
         exitInfo << "Database open error"
-                 << ", path=" << _name << endl;
+                 << ", path=" << _name << ", error_info:" << m_openStatus.ToString() << endl;
```
Comment (bxq2011hust) [inline libdevcore/BasicLevelDB.cpp]: use = replace : ,keep same

File: libblockchain/BlockChainImp.cpp [hunk 2]
```diff
@@ -315,11 +315,12 @@ void BlockChainImp::getNonces(
 {
     if (_blockNumber > number())
     {
-        BLOCKCHAIN_LOG(TRACE) << LOG_DESC("[#getNonces]Invalid block number")
+        BLOCKCHAIN_LOG(TRACE) << LOG_DESC("getNonces: Invalid block number")
```
Comment (bxq2011hust) [inline libblockchain/BlockChainImp.cpp]: what about delete ":" ?

File: libdevcore/BasicLevelDB.cpp (hunk 1 above)
Comment (cyjseagull) [reply libdevcore/BasicLevelDB.cpp]: done

File: libblockchain/BlockChainImp.cpp (hunk 2 above)
Comment (cyjseagull) [reply libblockchain/BlockChainImp.cpp]: done.

**Pull request merged**